### ISSUE TITLE
[NS-FS] Bucket is deleted but directory is still available under nsfs

### ISSUE THREAD

**Issue by cxhong** [noobaa/noobaa-core#6407]
My Kubenetes cluster:

```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:20:00Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
noobaa version
```
# noobaa version
INFO[0000] CLI version: 5.7.0-guy1-4e9c4bd
INFO[0000] noobaa-image: noobaa/noobaa-core:5.7.0-guy1-4e9c4bd
INFO[0000] operator-image: noobaa/noobaa-operator:5.7.0-guy1-4e9c4bd
```
I am running s3 command to create a bucket, put object, delete object, then delete the bucket.   At the end, bucket is deleted , but directory still available under nsfs
```
# s3 mb s3://test
make_bucket: test

# s3 ls
2021-03-10 06:22:40 test
2021-03-10 06:22:40 spark-s1
2021-03-10 06:22:40 presto-s1
2021-03-10 06:22:40 nsfs
2021-03-10 06:22:40 first.bucket
2021-03-10 06:22:40 buckettest
2021-03-10 06:22:40 bucket1-cached
# s3 rb s3://test
remove_bucket: test

# s3 ls
2021-03-10 11:38:09 spark-s1
2021-03-10 11:38:09 presto-s1
2021-03-10 11:38:09 nsfs
2021-03-10 11:38:09 first.bucket
2021-03-10 11:38:09 buckettest
2021-03-10 11:38:09 bucket1-cached
```
The dirs under nsfs
```
[root@hpotest-x-master pvc-2defee94-cbdd-4aab-8947-dd290a10bdb3-data]# pwd
/mnt/fs1/pvc-2defee94-cbdd-4aab-8947-dd290a10bdb3/pvc-2defee94-cbdd-4aab-8947-dd290a10bdb3-data
[root@hpotest-x-master pvc-2defee94-cbdd-4aab-8947-dd290a10bdb3-data]# ls -ltr
total 12
-rw-------  1 root  root    0 Mar  3 14:59 nohup.out
drwxr-xr-x  3 10001 root 4096 Mar  5 08:43 bucket1-cached
drwxrwxrwx 10 10001 root 4096 Mar  5 10:50 buckettest
drwxrwxrwx  4 root  root 4096 Mar  9 17:13 presto-s1
drwxr-xr-x  3 10001 root 4096 Mar  9 17:45 spark-s1
drwxr-xr-x  3 10001 root 4096 Mar 10 06:38 test          <----- still here
drwxr-xr-x  3 10001 root 4096 Mar 10 08:17 mycontainers1   <---- those two mycontainers dir is created from Cosbench, job is finished successfully, but dir still here.  issue #6272
drwxr-xr-x  3 10001 root 4096 Mar 10 08:17 mycontainers2
drwxrwxrwx 43 10001 root 8192 Mar 10 08:30 nsfs
```
The endpoint log didn't have any error message for 'DELETE'
```
Mar-10 19:36:56.587 [Endpoint/12]    [L0] core.util.signature_utils:: _string_to_sign_v4 method DELETE pathname /test search  headers { Host: '10.21.40.96:31042', 'Accept-Encoding': 'identity', 'User-Agent': 'aws-cli/2.1.25 Python/3.7.3 Linux/4.18.0-240.15.1.el8_3.x86_64 exe/x86_64.rhel.8 prompt/off command/s3.rb', 'X-Amz-Date': '20210310T193656Z', 'X-Amz-Content-Sha256': 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', Authorization: 'AWS4-HMAC-SHA256 Credential=L60n1kOE7WFbujukjCdQ/20210310/hpotest-x-app1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=54d3e313981d8f2eb74d771e4735b118c947da5bdb5802b4b7d4bce7d5280449', 'Content-Length': '0' } region hpotest-x-app1 canonical_str
DELETE
/test

host:10.21.40.96:31042
x-amz-content-sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date:20210310T193656Z

host;x-amz-content-sha256;x-amz-date
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
 string_to_sign
AWS4-HMAC-SHA256
20210310T193656Z
20210310/hpotest-x-app1/s3/aws4_request
7c32b8c33d3985546908196d14c3237ddb32f1710fc35dfe2cc9a27035e82c31

Mar-10 19:36:56.588 [Endpoint/12]    [L0] core.util.signature_utils:: _string_to_sign_v4 method DELETE pathname /test search  headers { Host: '10.21.40.96:31042', 'Accept-Encoding': 'identity', 'User-Agent': 'aws-cli/2.1.25 Python/3.7.3 Linux/4.18.0-240.15.1.el8_3.x86_64 exe/x86_64.rhel.8 prompt/off command/s3.rb', 'X-Amz-Date': '20210310T193656Z', 'X-Amz-Content-Sha256': 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', Authorization: 'AWS4-HMAC-SHA256 Credential=L60n1kOE7WFbujukjCdQ/20210310/hpotest-x-app1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=54d3e313981d8f2eb74d771e4735b118c947da5bdb5802b4b7d4bce7d5280449', 'Content-Length': '0' } region hpotest-x-app1 canonical_str
DELETE
```

**Comment by nimrod-becker** [comment]
We are aware of this and its an open task to be done

**Comment by nimrod-becker** [comment]
Fixed with #6466
